server/src/resources/cart-item/cart-item.repository.ts

```diff
@@ -0,0 +1,67 @@
+import { defaultCartItemDb } from '../../constants';
+import { CartItemDb } from '../../database/models.db';
+import { ICartItem, CartItemEntity } from './cart-item.interfaces';
+
+class CartItemRepository {
+  constructor() {
+    const addDefaultCartItems = async () => {
+      setTimeout(() => {
+        Promise.all(defaultCartItemDb.map(async (cartItem) => {
+          const found = await CartItemDb.findOne({
+            where: {
+              id: cartItem.id,
+            },
+          });
+          !found && await CartItemDb.create({ ...cartItem });
+        }));
+      }, 5000);
+    }
+    addDefaultCartItems();
+  }
+
+  getAllCartItems = async (): Promise<CartItemEntity[]> => {
+    const allCartItems = await CartItemDb.findAll() as unknown as CartItemEntity[];
```

Comment:
If you use generics when you define your db instances, such type assertions won't be needed.

Reply:
same is valid for all other requests